### DISCUSSION TITLE
Jaggluco connection with xdrip+

### DISCUSSION THREAD

**Discussion by Npavkov** [NightscoutFoundation/xDrip#2818] [Q&A] [answered]
Anyone out there able to get juggluco to send scan data to xdrip+. If so could you please share the setup settings for both juggluco and xdrip+.... It would be much appreciated.

**Comment by j-kaltes** [comment]
How to receive glucose values from Juggluco in xDrip is described in Juggluco Left menu->Settings->Help under "_Patched Libre Broadcast_".

**Comment by Npavkov** [comment]
does that help still work IF I am using Juggluco as the scanner for the sensor and NOT "Patched Libre Broadcast"?

**Comment by j-kaltes** [comment]
I didn't read correctly, you talked about "_scan data_". The only think you can do with scanned values is to save them in a file. Every else is done with the values received via Bluetooth.

**Comment by Npavkov** [comment]
I would really link to use DIABOX as the master collector and send the data to Juggluco then have Juggluco send the data out to xdrip+ !!!!

**Comment by j-kaltes** [comment]
Master collector for Bluetooth or scanned values? 
Juggluco can also send values received via Bluetooth to Diabox via the Nightscout server incorporated in Juggluco.

**Comment by Npavkov** [comment]
I would like to scan the sensor and receive Bluetooth and that seems more reliable connection to the sensor. then if Juggluco so it may send the data to libreview web site, And Also send the data on to xdrip+..
If that cant work, then how would I send data to diabox, or even xdrip+  thru the juggluco built in nightscout server (as you speak of above)???

There are some things I really like about Jugglco like the fact that it sends data easily to Libreview web, and the 1 minute sensor read updates. However the things I do not like about Juggluco are * Inconsistent Bluetooth connection with Libre2 sensor, and inability to figure out how to connect Juggluco with xdrip+.*

**Comment by j-kaltes** [comment]
To send data to Diabox, you need to turn on the web server in Juggluco, Left menu->Settings->Webserver->Active. The webserver in xDrip should be turned off before hand.
In Diabox you go to Settings. Set as Data Collection, Follower and give as url: http://127.0.0.1:17580

**Comment by Npavkov** [comment]
Juggluco asks for a api secret when I try to save the web server setting as active??

**Comment by j-kaltes** [comment]
You can keep it empty.

**Comment by Npavkov** [comment]
how do I set this:  In Diabox you go to Settings. Set as Data Collection, Follower and give as url: http://127.0.0.1:17580/ in diabox?

**Comment by Npavkov** [comment]
ok, I got diabox in follower mode.... ut I get no readings into juggluco..... does juggluco have a follower mode? so I can continue to use diabox to scan and get readings from the sensor?

**Comment by j-kaltes** [comment]
Juggluco can only receive glucose values directly from the sensor or via a mirror connection with Juggluco running on another device. You can scan with Diabox and Juggluco and receive the glucose values in Juggluco via Bluetooth from the sensor.

**Comment by Npavkov** [comment]
then I must have the sensor setup in Juggluco wrong??? how do I get Jugluco to scan the sensor again?

**Comment by j-kaltes** [comment]
Just scan.

**Comment by Npavkov** [comment]
ok, I scan in diabox, but I do not get data in juggluco?

**Comment by j-kaltes** [comment]
Scan some more in Juggluco, maybe wait and scan again. Force stop Diabox.

**Comment by Npavkov** [comment]
Ok, the scans seem to go into diabox and Juggluco ok, however NO Bluetooth readings yet?  I am now waiting to see if diabox gets bluetooth data or not..... Next step though, how do I set up Juggluco to send data to xdrip+?

**Comment by Npavkov** [comment]
Bluetooth data goes into diabox, but NOT juggluco now?

**Comment by j-kaltes** [comment]
What does Left menu->Sensor show? Is "_Use Bluetooth_" turned on?

**Comment by Npavkov** [comment]
yes..
it says it found the sensor.....but no Bluetooth data in Juggluco yet?

**Comment by j-kaltes** [comment]
Can you show a screenshot of Left menu->Sensor, what kind of error is present?

**Comment by Npavkov** [comment]
not sure how to get screenshot from android phone. I am on my computer communicating  with you now... screen shows last failures (status=8) alternates between last failure and last good reads.

**Comment by Npavkov** [comment]
still no bluetooth readings on Juggluco yet?

**Comment by j-kaltes** [comment]
Do you run Diabox and Juggluco on different devices? With European Freestyle Libre 2 sensors it is possible to run Diabox and Juggluco on the same device while both are connected to the sensor via Bluetooth (although you shouldn't do it because it increases the possibility of connection errors), but it is absolutely impossible when they run on different devices.
You should always force close Diabox.

**Comment by Npavkov** [comment]
Diabox seems to be working correctly. Juggluco sensor> shows a message that streaming not enabled,, could that be the issue?

**Comment by j-kaltes** [comment]
"Streaming not enabled" is not an issue. 
If you want to keep Diabox running, you should run it, but of you want to run Juggluco you have to force close it. 
And scan Juggluco a few times and maybe turn Bluetooth off and on and run Juggluco on your good phone.

**Comment by j-kaltes** [comment]
When you scan with Juggluco with "Sensor via Bluetooth" set, Juggluco will determine the authentication information needed to make use of the sensor via Bluetooth. 
Thereafter, Diabox can only get glucose values from the sensor via Bluetooth when Juggluco also connects with the sensor.
So it is only possible that you receive glucose values from Diabox via Bluetooth but not from Juggluco, when you later didn't scan with Juggluco or didn't do so often enough or you didn't have "Sensor via Bluetooth" set.

**Comment by Npavkov** [comment]
Jaap, is it possible to enter actual blood glucose values into Juggluco?

**Comment by j-kaltes** [comment]
One can enter numbers under whatever label you want, with Left middle menu->New Amount. One can determine the labels in Left menu->Settings->Number Labels.

**Comment by Npavkov** [comment]
I do not see a “new Amounts” option under middle menu???

**Reply by Navid200** [reply]
@Npavkov 
Please don't reply to an email in order to post.  Even though, from your perspective, it may work out fine, if you log into the discussions and look at the thread, you can see how impractical it becomes for anyone to go through the thread.  These discussions should be useful to everyone even those who may come to a thread through a search online.

**Comment by j-kaltes** [comment]
![NewAmount](https://user-images.githubusercontent.com/72013079/236640443-cba90d78-f3a9-4c1f-a42d-01992f69fbe0.jpg)

**Comment by Npavkov** [comment]
I see ALL the other options on that menu EXCEPT “New Amount” option. It is NOT there?

**Comment by j-kaltes** [comment]
O, then you have set "_Amounts received from mirror_" or so in Left middle menu->Mirror.

**Reply by Npavkov** [reply]
Does entering the "new Amount" function have any effect on the actual bluetooth reading of the sensor (using Juggluco)? Or ANY calibration done by Juggluco using the "new Amount"  values entered?

**Reply by j-kaltes** [reply, marked as answer]
No.

**Comment by Npavkov** [comment]
OK, Gotcha…. I see it now, Thanks…. Last question (at least for today….Is there anyway to get xdrip+ to send readings to nightscout every minute, instead of every 5 minutes?

**Comment by Npavkov** [comment]
OK, Gotcha…. I see it now, Thanks…. Last question (at least for today….Is there anyway to get xdrip+ to send readings to nightscout every minute, instead of every 5 minutes?

Wait, I guess I should ask....Does  juggluco send the readings out to webserver as it receives them (every Minute)?

**Comment by j-kaltes** [comment]
In Juggluco, you can do so in Left menu->Settings->Uploader.

**Comment by Npavkov** [comment]
I used the Juggluco "web Server" option. I tried uploader, however I did not think it worked in sending data to xfrip+ at all?
I took it out of uploader, and put it in webserver, and then xdrip+ started getting the data.

**Comment by j-kaltes** [comment]
Uploader uploads the glucose values to a Nightscout server.
The frequency of access to the web server is determined by receiver. xDrip or another Nightscout client asks for those values and receives immediately the most recent values.

**Comment by Npavkov** [comment]
ok, so you are saying that xdrip+ (nightscout Follower) requests the data every 5 mins?

**Comment by j-kaltes** [comment]
Yes or even less frequently. Some other nightscout clients can ask more frequently.

**Comment by Npavkov** [comment]
so there is no way to get xdrip to ask for the readings every minute as they are available?

**Comment by j-kaltes** [comment]
https://github.com/NightscoutFoundation/xDrip/issues/2680

**Comment by Npavkov** [comment]
OK, I have that option Libre 1 minute checked, but it does not seem to work yet with Nightscout follower as input to xdrip+?  :(

will that be planned in a futre version of xdrip+?

**Comment by Npavkov** [comment]
I want to thank you guys for ALL your help you have given me already. I could not have gotten this far without it. I AT LEAST have gotten Juggluco to accept nfc scans and bluetooth data from my  libre2 US sensor, and sending data to libreview (for my doctors) and also sending the data to xdrip+ so it can upload to my personal nightscout de account. Thank You SOOOOO Much!!
